client/runtime/src/main/java/io/quarkiverse/openapi/generator/providers/AbstractAuthProvider.java

```diff
@@ -60,22 +60,33 @@ public String getName() {
     }
 
     public boolean isTokenPropagation() {
-        return ConfigProvider.getConfig()
-                .getOptionalValue(getCanonicalAuthConfigPropertyName(AuthConfig.TOKEN_PROPAGATION), Boolean.class)
-                .orElse(false);
+        return isTokenPropagation(getOpenApiSpecId(), getName());
     }
 
-    public String getTokenForPropagation(MultivaluedMap<String, Object> httpHeaders) {
-        String headerName = getHeaderName() != null ? getHeaderName() : HttpHeaders.AUTHORIZATION;
-        String propagatedHeaderName = propagationHeaderName(getOpenApiSpecId(), getName(), headerName);
+    public static String getTokenForPropagation(MultivaluedMap<String, Object> httpHeaders, String openApiSpecId,
+            String authName) {
+        String headerName = getHeaderName(openApiSpecId, authName) != null ? getHeaderName(openApiSpecId, authName)
```

Comment:
Why not just `getName()` instead of receiving the `authName`? Something doesn't seem right. Maybe we should remove this method from here and move it to a utility class instead.

Reply:
that's because the `authName` is not a static field and we are in a static method

> Maybe we should remove this method from here and move it to a utility class instead.

Yes as this is used in different places